main/src/ca/uwaterloo/flix/api/lsp/provider/completion/EnumCompleter.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2022 Paul Butcher, Lukas Rønn
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.api.lsp.provider.completion
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.api.lsp.provider.completion.Completion.EnumCompletion
+import ca.uwaterloo.flix.api.lsp.provider.completion.CompletionUtils.mkModuleSym
+import ca.uwaterloo.flix.api.lsp.provider.completion.TypeCompleter.{formatTParams, formatTParamsSnippet, getInternalPriority, priorityBoostForTypes}
+import ca.uwaterloo.flix.api.lsp.{Index, TextEdit}
+import ca.uwaterloo.flix.language.ast.Symbol.{EnumSym, ModuleSym}
+import ca.uwaterloo.flix.language.ast.{Symbol, TypedAst}
+
+object EnumCompleter extends Completer {
+  /**
+    * Returns a List of Completion for enums.
+    */
+  override def getCompletions(context: CompletionContext)(implicit flix: Flix, index: Index, root: TypedAst.Root, delta: DeltaContext): Iterable[EnumCompletion] = {
+    // We use a fqn to lookup in the modules
+    // We therefore need to split the context.word at dots
+    val word = context.word.split('.').toList
+    val (fqn, subWord) = {
```

Comment:
Why all this subword. Can we not let VSCode take care of it?

Reply:
I would think all of these cases could become one.

Reply:
No, we cannot let VSCode take care of it, because then the `Ranker` won't work as intended.